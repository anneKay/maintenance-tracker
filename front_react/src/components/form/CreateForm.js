import React, { Component } from 'react';
import {
  Form, Message, Dropdown, Label 
} from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import sendRequest from './CreateReqHOC';
import FormField from './InputFormField';
import { options } from '../../helpers/helper';

export const CreateForm = ({
  onChange, onSubmit, state, request, userRequest,
}) => {
  const {
    loading,
    data: {
      title,
    },
    errors,
  } = state;

  return (
    <div>
      {request && console.log('>>>>>>>requesssinedit', request)}
      <main>
        <Form onSubmit={onSubmit}>
          {FormField('text', 'title', onChange, title, 'Enter your request title', 'Title:', errors.title)}
          <Form.Field error={!!errors.requestType}>
            <label htmlFor="requestType">Request Type</label>
            <Dropdown
              placeholder="select request type"
              fluid
              search
              selection
              options={options}
              onChange={onChange}
            />
            {errors.requestType && (
            <Label color="red" pointing>
              {errors.requestType}
            </Label>)}
          </Form.Field>
          <Form.Field error={!!errors.description}>
            <label htmlFor="decription">Description</label>
            <textarea
              required
              rows={4}
              cols={50}
              type="text"
              id="description"
              name="description"
              placeholder="Describe your request"
              onChange={onChange}
            >
              {userRequest && userRequest[0] ? userRequest[0].description : null}
            </textarea>
            {errors.description && (
              <Label color="red" pointing>
                {errors.description}
              </Label>)}
          </Form.Field>
          <input type="submit" value="Submit" />
          <Link to="/profile"><input value="Cancel" id="cancel" /></Link>
        </Form>
      </main>
    </div>
  );
};

CreateForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  state: PropTypes.shape({

  }).isRequired,

};

export default sendRequest(CreateForm);
