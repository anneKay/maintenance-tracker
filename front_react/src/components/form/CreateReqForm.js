import React from 'react';
import { Form,  Message, Dropdown, Label } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { options } from '../../helpers/helper';

class CreateReq extends React.Component {
    state = {
      loading: false,
			data:{
				title: '',
        description: '',
      },
      requestType: '',
			errors:{}
    };
    
		onChange = (event, dropdown) => {
      const { data } = this.state;
		this.setState({ 
      errors: {},
      data: {
        ...data,
        [event.target.name]: event.target.value,
      },
    })
    if(dropdown) {
      this.setState({
        requestType: dropdown.value
      })
    }
  }

		validate = (data, requestType) => {
			const errors = {};
      if (data.title.trim().length <= 4) errors.title = 'Add a valid title';
      if (data.description.trim().length <= 10 ) errors.description = 'Description should not be less than 8 characters';
      if (requestType.trim().length === 0 ) errors.requestType = 'Please select a request type'; 
			return errors;
		}

		onSubmit = () => {
      const { submit, history } = this.props;
      const { data, requestType } = this.state;
			const errors = this.validate(data, requestType);
			this.setState({
				errors,
			})
			if(Object.keys(errors).length === 0 ){
			submit(data, requestType, history);
			}
		}

    render() {
      const {data, errors} = this.state;
			return (
        <div>
          <main>
            <Form onSubmit={this.onSubmit}>
              <Form.Field error={!!errors.title}>
                  <label htmlFor="title">Title</label>
                  <input
                  required
                  type="text" 
                  id="title" 
                  name="title" 
                  placeholder="Enter your request title"
                  value={data.title}
                  onChange={this.onChange}
                  />
                  {errors.title && (
                  <Label color="red" pointing>
                  {errors.title}
                  </Label>)}
              </Form.Field>
              <Form.Field error={!!errors.requestType}>
                <label htmlFor="requestType">Request Type</label>
                <Dropdown 
                  placeholder='select request type'
                  fluid
                  search 
                  selection
                  options={options}
                  onChange={this.onChange} />
                   {errors.requestType && (
                  <Label color="red" pointing>
                  {errors.requestType}
                  </Label>)}
              </Form.Field>
              <Form.Field error={!!errors.description}> 
                  <label htmlFor="decription">Description</label>
                  <textarea
                  required
                  rows={4}
                  cols={50}
                  type="text" 
                  id="description" 
                  name="description" 
                  placeholder="Descibe your request"
                  onChange={this.onChange}>
                  </textarea>
                  {errors.description && (
                  <Label color="red" pointing>
                  {errors.description}
                  </Label>)}
              </Form.Field>
					<input type="submit"  value="Submit"/>
          <Link to="/profile"><input value="Cancel" id="cancel"/></Link>
        </Form>
      </main>
		</div>
			)
    }  
}
CreateReq.propTypes = {
	submit: PropTypes.func.isRequired
};

export default CreateReq;