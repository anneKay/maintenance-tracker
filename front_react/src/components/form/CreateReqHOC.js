import React, { Component } from 'react';

const sendRequest = (WrappedComponent) => {
  class HOC extends Component {
    state = {
      loading: false,
      data: {
        title: '',
        description: '',
      },
      requestType: '',
      errors: {},
    };

    onChange = (event, dropdown) => {
      const { data } = this.state;
      this.setState({
        errors: {},
        data: {
          ...data,
          [event.target.name]: event.target.value,
        },
      });
      if (dropdown) {
        this.setState({
          requestType: dropdown.value,
        });
      }
    }

validate = (data, requestType) => {
  const errors = {};
  if (data.title.trim().length <= 4) errors.title = 'Add a valid title';
  if (data.description.trim().length <= 10) errors.description = 'Description should not be less than 8 characters';
  if (requestType.trim().length === 0) errors.requestType = 'Please select a request type';
  return errors;
}

onSubmit = () => {
  const { submit, history } = this.props;
  const { data, requestType } = this.state;
  const errors = this.validate(data, requestType);
  this.setState({
    errors,
  });
  if (Object.keys(errors).length === 0) {
    submit(data, requestType, history);
  }
}

render() {
  return (
    <WrappedComponent
      onChange={this.onChange}
      validate={this.validate}
      onSubmit={this.onSubmit}
      state={this.state}
      {...this.props}
    />
  );
}
  }
  return HOC;
};
export default sendRequest;
