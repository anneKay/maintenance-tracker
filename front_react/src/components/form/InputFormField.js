import React from 'react';
import { Form, Label } from 'semantic-ui-react';

const FormField = (type, name, onChange, value, placeholder, label, errors, defaultValue) => (
  <Form.Field error={!!errors}>
    <label htmlFor={name}>
      {label}
      <input
        required
        defaultValue={defaultValue}
        type={type}
        id={name}
        name={name}
        onChange={onChange}
        value={value}
        placeholder={placeholder}
      />
    </label>
    {errors && (
      <Label color="red" pointing>
        {errors}
      </Label>
    )}
  </Form.Field>
);

export default FormField;
