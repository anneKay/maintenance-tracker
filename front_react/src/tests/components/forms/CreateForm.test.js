import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { CreateForm } from '../../../components/form/CreateForm';

describe(' Login Form component rendering', () => {
  const state = {
    loading: false,
    data: {
      title: '',
      description: '',
    },
    requestType: '',
    errors: {},
  };
  const sinonSpy = sinon.spy();
  const wrapper = shallow(<CreateForm
    state={state}
    validate={sinonSpy}
    onSubmit={sinonSpy}
    onChange={sinonSpy}
  />);

  it('Should render properly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call the submit method on submit', () => {
    wrapper.find('Form').simulate('submit', { preventDefault() { } });
    expect(sinonSpy.called).toBe(true);
  });
});
