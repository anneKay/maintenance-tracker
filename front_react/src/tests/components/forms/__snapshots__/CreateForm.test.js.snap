// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Login Form component rendering Should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateForm
    onChange={[Function]}
    onSubmit={[Function]}
    state={
      Object {
        "data": Object {
          "description": "",
          "title": "",
        },
        "errors": Object {},
        "loading": false,
        "requestType": "",
      }
    }
    validate={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main>
        <Form
          as="form"
          onSubmit={[Function]}
        >
          <FormField
            error={false}
          >
            <label
              htmlFor="title"
            >
              Title:
              <input
                id="title"
                name="title"
                onChange={[Function]}
                placeholder="Enter your request title"
                required={true}
                type="text"
                value=""
              />
            </label>
          </FormField>
          <FormField
            error={false}
          >
            <label
              htmlFor="requestType"
            >
              Request Type
            </label>
            <Dropdown
              additionLabel="Add "
              additionPosition="top"
              closeOnBlur={true}
              deburr={false}
              fluid={true}
              icon="dropdown"
              minCharacters={1}
              noResultsMessage="No results found."
              onChange={[Function]}
              openOnFocus={true}
              options={
                Array [
                  Object {
                    "key": "urgent",
                    "text": "urgent",
                    "value": "urgent",
                  },
                  Object {
                    "key": "normal",
                    "text": "normal",
                    "value": "normal",
                  },
                  Object {
                    "key": "nonurgent",
                    "text": "Not urgent",
                    "value": "nonurgent",
                  },
                ]
              }
              placeholder="select request type"
              renderLabel={[Function]}
              search={true}
              searchInput="text"
              selectOnBlur={true}
              selectOnNavigation={true}
              selection={true}
              wrapSelection={true}
            />
          </FormField>
          <FormField
            error={false}
          >
            <label
              htmlFor="decription"
            >
              Description
            </label>
            <textarea
              cols={50}
              id="description"
              name="description"
              onChange={[Function]}
              placeholder="Describe your request"
              required={true}
              rows={4}
              type="text"
            />
          </FormField>
          <input
            type="submit"
            value="Submit"
          />
          <Link
            replace={false}
            to="/profile"
          >
            <input
              id="cancel"
              value="Cancel"
            />
          </Link>
        </Form>
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          as="form"
          onSubmit={[Function]}
        >
          <FormField
            error={false}
          >
            <label
              htmlFor="title"
            >
              Title:
              <input
                id="title"
                name="title"
                onChange={[Function]}
                placeholder="Enter your request title"
                required={true}
                type="text"
                value=""
              />
            </label>
          </FormField>
          <FormField
            error={false}
          >
            <label
              htmlFor="requestType"
            >
              Request Type
            </label>
            <Dropdown
              additionLabel="Add "
              additionPosition="top"
              closeOnBlur={true}
              deburr={false}
              fluid={true}
              icon="dropdown"
              minCharacters={1}
              noResultsMessage="No results found."
              onChange={[Function]}
              openOnFocus={true}
              options={
                Array [
                  Object {
                    "key": "urgent",
                    "text": "urgent",
                    "value": "urgent",
                  },
                  Object {
                    "key": "normal",
                    "text": "normal",
                    "value": "normal",
                  },
                  Object {
                    "key": "nonurgent",
                    "text": "Not urgent",
                    "value": "nonurgent",
                  },
                ]
              }
              placeholder="select request type"
              renderLabel={[Function]}
              search={true}
              searchInput="text"
              selectOnBlur={true}
              selectOnNavigation={true}
              selection={true}
              wrapSelection={true}
            />
          </FormField>
          <FormField
            error={false}
          >
            <label
              htmlFor="decription"
            >
              Description
            </label>
            <textarea
              cols={50}
              id="description"
              name="description"
              onChange={[Function]}
              placeholder="Describe your request"
              required={true}
              rows={4}
              type="text"
            />
          </FormField>
          <input
            type="submit"
            value="Submit"
          />
          <Link
            replace={false}
            to="/profile"
          >
            <input
              id="cancel"
              value="Cancel"
            />
          </Link>
        </Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "form",
          "children": Array [
            <FormField
              error={false}
            >
              <label
                htmlFor="title"
              >
                Title:
                <input
                  id="title"
                  name="title"
                  onChange={[Function]}
                  placeholder="Enter your request title"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
            </FormField>,
            <FormField
              error={false}
            >
              <label
                htmlFor="requestType"
              >
                Request Type
              </label>
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                fluid={true}
                icon="dropdown"
                minCharacters={1}
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "key": "urgent",
                      "text": "urgent",
                      "value": "urgent",
                    },
                    Object {
                      "key": "normal",
                      "text": "normal",
                      "value": "normal",
                    },
                    Object {
                      "key": "nonurgent",
                      "text": "Not urgent",
                      "value": "nonurgent",
                    },
                  ]
                }
                placeholder="select request type"
                renderLabel={[Function]}
                search={true}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                selection={true}
                wrapSelection={true}
              />
            </FormField>,
            <FormField
              error={false}
            >
              <label
                htmlFor="decription"
              >
                Description
              </label>
              <textarea
                cols={50}
                id="description"
                name="description"
                onChange={[Function]}
                placeholder="Describe your request"
                required={true}
                rows={4}
                type="text"
              />
            </FormField>,
            <input
              type="submit"
              value="Submit"
            />,
            <Link
              replace={false}
              to="/profile"
            >
              <input
                id="cancel"
                value="Cancel"
              />
            </Link>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="title"
                >
                  Title:
                  <input
                    id="title"
                    name="title"
                    onChange={[Function]}
                    placeholder="Enter your request title"
                    required={true}
                    type="text"
                    value=""
                  />
                </label>,
                undefined,
              ],
              "error": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Title:",
                    <input
                      id="title"
                      name="title"
                      onChange={[Function]}
                      placeholder="Enter your request title"
                      required={true}
                      type="text"
                      value=""
                    />,
                  ],
                  "htmlFor": "title",
                },
                "ref": null,
                "rendered": Array [
                  "Title:",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "defaultValue": undefined,
                      "id": "title",
                      "name": "title",
                      "onChange": [Function],
                      "placeholder": "Enter your request title",
                      "required": true,
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "label",
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="requestType"
                >
                  Request Type
                </label>,
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  closeOnBlur={true}
                  deburr={false}
                  fluid={true}
                  icon="dropdown"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "key": "urgent",
                        "text": "urgent",
                        "value": "urgent",
                      },
                      Object {
                        "key": "normal",
                        "text": "normal",
                        "value": "normal",
                      },
                      Object {
                        "key": "nonurgent",
                        "text": "Not urgent",
                        "value": "nonurgent",
                      },
                    ]
                  }
                  placeholder="select request type"
                  renderLabel={[Function]}
                  search={true}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  selection={true}
                  wrapSelection={true}
                />,
                undefined,
              ],
              "error": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Request Type",
                  "htmlFor": "requestType",
                },
                "ref": null,
                "rendered": "Request Type",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "additionLabel": "Add ",
                  "additionPosition": "top",
                  "closeOnBlur": true,
                  "deburr": false,
                  "fluid": true,
                  "icon": "dropdown",
                  "minCharacters": 1,
                  "noResultsMessage": "No results found.",
                  "onChange": [Function],
                  "openOnFocus": true,
                  "options": Array [
                    Object {
                      "key": "urgent",
                      "text": "urgent",
                      "value": "urgent",
                    },
                    Object {
                      "key": "normal",
                      "text": "normal",
                      "value": "normal",
                    },
                    Object {
                      "key": "nonurgent",
                      "text": "Not urgent",
                      "value": "nonurgent",
                    },
                  ],
                  "placeholder": "select request type",
                  "renderLabel": [Function],
                  "search": true,
                  "searchInput": "text",
                  "selectOnBlur": true,
                  "selectOnNavigation": true,
                  "selection": true,
                  "wrapSelection": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="decription"
                >
                  Description
                </label>,
                <textarea
                  cols={50}
                  id="description"
                  name="description"
                  onChange={[Function]}
                  placeholder="Describe your request"
                  required={true}
                  rows={4}
                  type="text"
                />,
                undefined,
              ],
              "error": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Description",
                  "htmlFor": "decription",
                },
                "ref": null,
                "rendered": "Description",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "cols": 50,
                  "id": "description",
                  "name": "description",
                  "onChange": [Function],
                  "placeholder": "Describe your request",
                  "required": true,
                  "rows": 4,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <input
                id="cancel"
                value="Cancel"
              />,
              "replace": false,
              "to": "/profile",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "cancel",
                "value": "Cancel",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <Form
            as="form"
            onSubmit={[Function]}
          >
            <FormField
              error={false}
            >
              <label
                htmlFor="title"
              >
                Title:
                <input
                  id="title"
                  name="title"
                  onChange={[Function]}
                  placeholder="Enter your request title"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
            </FormField>
            <FormField
              error={false}
            >
              <label
                htmlFor="requestType"
              >
                Request Type
              </label>
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                fluid={true}
                icon="dropdown"
                minCharacters={1}
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "key": "urgent",
                      "text": "urgent",
                      "value": "urgent",
                    },
                    Object {
                      "key": "normal",
                      "text": "normal",
                      "value": "normal",
                    },
                    Object {
                      "key": "nonurgent",
                      "text": "Not urgent",
                      "value": "nonurgent",
                    },
                  ]
                }
                placeholder="select request type"
                renderLabel={[Function]}
                search={true}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                selection={true}
                wrapSelection={true}
              />
            </FormField>
            <FormField
              error={false}
            >
              <label
                htmlFor="decription"
              >
                Description
              </label>
              <textarea
                cols={50}
                id="description"
                name="description"
                onChange={[Function]}
                placeholder="Describe your request"
                required={true}
                rows={4}
                type="text"
              />
            </FormField>
            <input
              type="submit"
              value="Submit"
            />
            <Link
              replace={false}
              to="/profile"
            >
              <input
                id="cancel"
                value="Cancel"
              />
            </Link>
          </Form>
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Form
            as="form"
            onSubmit={[Function]}
          >
            <FormField
              error={false}
            >
              <label
                htmlFor="title"
              >
                Title:
                <input
                  id="title"
                  name="title"
                  onChange={[Function]}
                  placeholder="Enter your request title"
                  required={true}
                  type="text"
                  value=""
                />
              </label>
            </FormField>
            <FormField
              error={false}
            >
              <label
                htmlFor="requestType"
              >
                Request Type
              </label>
              <Dropdown
                additionLabel="Add "
                additionPosition="top"
                closeOnBlur={true}
                deburr={false}
                fluid={true}
                icon="dropdown"
                minCharacters={1}
                noResultsMessage="No results found."
                onChange={[Function]}
                openOnFocus={true}
                options={
                  Array [
                    Object {
                      "key": "urgent",
                      "text": "urgent",
                      "value": "urgent",
                    },
                    Object {
                      "key": "normal",
                      "text": "normal",
                      "value": "normal",
                    },
                    Object {
                      "key": "nonurgent",
                      "text": "Not urgent",
                      "value": "nonurgent",
                    },
                  ]
                }
                placeholder="select request type"
                renderLabel={[Function]}
                search={true}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
                selection={true}
                wrapSelection={true}
              />
            </FormField>
            <FormField
              error={false}
            >
              <label
                htmlFor="decription"
              >
                Description
              </label>
              <textarea
                cols={50}
                id="description"
                name="description"
                onChange={[Function]}
                placeholder="Describe your request"
                required={true}
                rows={4}
                type="text"
              />
            </FormField>
            <input
              type="submit"
              value="Submit"
            />
            <Link
              replace={false}
              to="/profile"
            >
              <input
                id="cancel"
                value="Cancel"
              />
            </Link>
          </Form>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "form",
            "children": Array [
              <FormField
                error={false}
              >
                <label
                  htmlFor="title"
                >
                  Title:
                  <input
                    id="title"
                    name="title"
                    onChange={[Function]}
                    placeholder="Enter your request title"
                    required={true}
                    type="text"
                    value=""
                  />
                </label>
              </FormField>,
              <FormField
                error={false}
              >
                <label
                  htmlFor="requestType"
                >
                  Request Type
                </label>
                <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  closeOnBlur={true}
                  deburr={false}
                  fluid={true}
                  icon="dropdown"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  onChange={[Function]}
                  openOnFocus={true}
                  options={
                    Array [
                      Object {
                        "key": "urgent",
                        "text": "urgent",
                        "value": "urgent",
                      },
                      Object {
                        "key": "normal",
                        "text": "normal",
                        "value": "normal",
                      },
                      Object {
                        "key": "nonurgent",
                        "text": "Not urgent",
                        "value": "nonurgent",
                      },
                    ]
                  }
                  placeholder="select request type"
                  renderLabel={[Function]}
                  search={true}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
                  selection={true}
                  wrapSelection={true}
                />
              </FormField>,
              <FormField
                error={false}
              >
                <label
                  htmlFor="decription"
                >
                  Description
                </label>
                <textarea
                  cols={50}
                  id="description"
                  name="description"
                  onChange={[Function]}
                  placeholder="Describe your request"
                  required={true}
                  rows={4}
                  type="text"
                />
              </FormField>,
              <input
                type="submit"
                value="Submit"
              />,
              <Link
                replace={false}
                to="/profile"
              >
                <input
                  id="cancel"
                  value="Cancel"
                />
              </Link>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="title"
                  >
                    Title:
                    <input
                      id="title"
                      name="title"
                      onChange={[Function]}
                      placeholder="Enter your request title"
                      required={true}
                      type="text"
                      value=""
                    />
                  </label>,
                  undefined,
                ],
                "error": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Title:",
                      <input
                        id="title"
                        name="title"
                        onChange={[Function]}
                        placeholder="Enter your request title"
                        required={true}
                        type="text"
                        value=""
                      />,
                    ],
                    "htmlFor": "title",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Title:",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "defaultValue": undefined,
                        "id": "title",
                        "name": "title",
                        "onChange": [Function],
                        "placeholder": "Enter your request title",
                        "required": true,
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "label",
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="requestType"
                  >
                    Request Type
                  </label>,
                  <Dropdown
                    additionLabel="Add "
                    additionPosition="top"
                    closeOnBlur={true}
                    deburr={false}
                    fluid={true}
                    icon="dropdown"
                    minCharacters={1}
                    noResultsMessage="No results found."
                    onChange={[Function]}
                    openOnFocus={true}
                    options={
                      Array [
                        Object {
                          "key": "urgent",
                          "text": "urgent",
                          "value": "urgent",
                        },
                        Object {
                          "key": "normal",
                          "text": "normal",
                          "value": "normal",
                        },
                        Object {
                          "key": "nonurgent",
                          "text": "Not urgent",
                          "value": "nonurgent",
                        },
                      ]
                    }
                    placeholder="select request type"
                    renderLabel={[Function]}
                    search={true}
                    searchInput="text"
                    selectOnBlur={true}
                    selectOnNavigation={true}
                    selection={true}
                    wrapSelection={true}
                  />,
                  undefined,
                ],
                "error": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Request Type",
                    "htmlFor": "requestType",
                  },
                  "ref": null,
                  "rendered": "Request Type",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "additionLabel": "Add ",
                    "additionPosition": "top",
                    "closeOnBlur": true,
                    "deburr": false,
                    "fluid": true,
                    "icon": "dropdown",
                    "minCharacters": 1,
                    "noResultsMessage": "No results found.",
                    "onChange": [Function],
                    "openOnFocus": true,
                    "options": Array [
                      Object {
                        "key": "urgent",
                        "text": "urgent",
                        "value": "urgent",
                      },
                      Object {
                        "key": "normal",
                        "text": "normal",
                        "value": "normal",
                      },
                      Object {
                        "key": "nonurgent",
                        "text": "Not urgent",
                        "value": "nonurgent",
                      },
                    ],
                    "placeholder": "select request type",
                    "renderLabel": [Function],
                    "search": true,
                    "searchInput": "text",
                    "selectOnBlur": true,
                    "selectOnNavigation": true,
                    "selection": true,
                    "wrapSelection": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="decription"
                  >
                    Description
                  </label>,
                  <textarea
                    cols={50}
                    id="description"
                    name="description"
                    onChange={[Function]}
                    placeholder="Describe your request"
                    required={true}
                    rows={4}
                    type="text"
                  />,
                  undefined,
                ],
                "error": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                    "htmlFor": "decription",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                    "cols": 50,
                    "id": "description",
                    "name": "description",
                    "onChange": [Function],
                    "placeholder": "Describe your request",
                    "required": true,
                    "rows": 4,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <input
                  id="cancel"
                  value="Cancel"
                />,
                "replace": false,
                "to": "/profile",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "cancel",
                  "value": "Cancel",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
